doc/README-mosquitto.md

```diff
@@ -0,0 +1,151 @@
+
+```
+❯ ls
+sha512_hashes     HMAC_hashes
+# Identify the SHA512 hashes by their format: HEX_DIGEST:HEX_SALT
+❯ head -n 1 sha512_hashes
+SOMELONGHEXDIGEST:SOMEHEXSALT
+❯ hashcat -a 0 -m 1710 --hex-salt ./sha512_hashes [WORDLIST]
+[... SNIP ...]
+# Identify the HMAC hashes by a leading sha512: and following iteration count
+# These also have base64 salt and digests
+❯ head -n 1 HMAC_hashes
+sha512:101:BaSe64SaLt:BaSe64DiGeSt==
+❯ hashcat -a 0 -m 12100 ./HMAC_hashes [WORDLIST]
+[... SNIP ...]
+```
+
+**Note** the *use of --hex-salt in mode 1710* this is required for successful
+cracking.
+
+## Troubleshooting
+
+### No hashes loaded?
+
+If your individual install isn't automatically detecting the hash type, you 
+can try manually checking the hash format and specifying it with *--format*
+on the John command line:
+
+```
+# The format will be just after the username field
+❯ head -n 1 ./mosquitto.hash
+username:$dynamic_82$SOMELONGHASH$SOMESALT	
+# Now manually specify the dynamic_82 format
+❯ john ./mosquitto.hash --format=dynamic_82
+[...SNIP...]
+Session completed. 
+
+# Again for the pbkdf2-hmac-sha512 format
+❯ head -n 1 ./different_mosquitto.hash
+username2:$pbkdf2-hmac-sha512$101$.SOMESALT.SOMELONGHASH	
+# Now manually specify the pbkdf2-hmac-sha512 format
+❯ john ./mosquitto.hash --format=pbkdf2-hmac-sha512
+[...SNIP...]
+Session completed. 
+```
+
+### Mixed Hash Types?
+
+The Eclipse mosquitto_passwd program might allow different hash varieties, 
+depending on the Version you're working against. In the rare case that this
+could come up, you'll see an error and will be prompted to specify an 
+`out_file_name`; the script will then split the hash types out for you to 
+crack independently from one another. Note that the file extensions 
+allude to the hash format, handy if you want to use the --format flag in
+John. 
+
+```
+❯ mosquitto2john.py mosquitto_passwd > mosquitto.hash
+./mqt_pwd contains mixed hash types
+Specify an out_file_name to split these out.
+❯ mosquitto2john.py mosquitto_passwd mosquitto_out
+❯ ls
+mosquitto_out.dynamic_82      mosquitto_out.pbkdf2-hmac-sha512
+# Now crack one at a time because different hash modes
+❯ john ./mosquitto_out.dynamic_82 
+[... SNIP ...]
+❯ john ./mosquitto_out.pbkdf2-hmac-sha512
+[... SNIP ...]
+```
+
+And you're done! If the hashes aren't being detected automatically, see
+the section above.
+
+Similarly, if using Hashcat, we need to crack any mixed hash formats one 
+flavour at a time because different hash modes. Note in the following example
+that the outfiles have an extension for the hashcat hash mode in question.
+
+```
+❯ ls
+mosquitto_out.1710      mosquitto_out.12100
+❯ hashcat -a 0 -m 1710 --hex-salt ./hcat_out_file.1710.hcat [WORDLIST]
+[... SNIP ...]
+❯ hashcat -a 0 -m 12100 ./hcat_out_file.12100.hcat [WORDLIST]
+[... SNIP ...]
+```
+Again, you need to **specify --hex-salt** for the 1710 format hashes.
+
+### Invlid Input. Try removing ':' ?
```

Comment:
Typo in "Invlid". Should be the same as what the script actually says, which is: "Invalid input. Try removing ':' from username".